apps/backend/message/src/index.ts

```diff
@@ -11,16 +15,34 @@ app.use(express.json());
 app.use(cookieParser());
 app.use(authenticate_request);
 
+const pgp = pgPromise();
+const DB_ENDPOINT = process.env.DB_ENDPOINT;
+if (!DB_ENDPOINT) {
+  throw new Error("DB_ENDPOINT is not set");
+}
+
+const db = pgp(DB_ENDPOINT);
+
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 app.use((err: Error, req: Request, res: Response, next: NextFunction) => {
   console.error(err.stack);
   res.status(500).send("Something went wrong");
 });
 
-app.get("/api/messages", (req: Request, res: Response) => {
-  res.send("Hello world");
-});
+app.post("/api/messages", useValidateCreateMessage, (req, res) =>
+  createMessage(req, res, db),
+);
+app.get(
+  "/api/messages/thread/:listing_id/:receiver_id",
+  useValidateGetMessages,
+  (req, res) => getMessages(req, res, db),
+);
+app.get("/api/messages/overview", (req, res) =>
+  getMessageThreads(req, res, db),
+);
```

Comment:
All of these `req` will need to be typecast to `AuthenticatedRequest`. It's gross, but take a look at the other backend services